batch_requests.go

```diff
@@ -0,0 +1,94 @@
+package msgraphgocore
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io"
+	"net/http"
+
+	"github.com/google/uuid"
+	abstractions "github.com/microsoft/kiota/abstractions/go"
+)
+
+type batchItem struct {
+	Id        string            `json:"id"`
+	Method    string            `json:"method"`
+	Url       string            `json:"url"`
+	Headers   map[string]string `json:"headers"`
+	Body      string            `json:"body"`
+	DependsOn []string          `json:"dependsOn"`
+}
+
+func newBatchItem(requestInfo abstractions.RequestInformation) (*batchItem, error) {
```

Comment:
use a pointer instead please, not only because the fluent API returns a pointer but also to avoid the raw url issue on getURI.
https://github.com/microsoftgraph/msgraph-sdk-go/blob/7d7ebff89500456fd466788c39810d2f41bb54b3/me/me_request_builder.go#L322